### PR TITLE
fix base(b, big(0), 0) == "0"

### PR DESCRIPTION
**Pull request by rfourquet**
When `pad==0`, `base` should give `""` on input `big(0)`.
I merged the two versions `base(b, ::BigInt)` and `base(b, ::BigInt, pad)`. Without `pad`, it gets about 10% slower (so I could re-add the old version if requested), but the version with a pad is vastly faster (it gets faster when `pad` gets bigger).
Also I'm not sure: the buffer sent to libgmp must contain space for a final `'\0'`, do I need to handle that manually or `Sting` allocates automatically 1 byte for it?

### CODE REVIEW

**Comment by rfourquet** [comment]
Bump (CI failure is unrelated).

**Comment by rfourquet** [comment]
bump

**Review by StefanKarpinski — APPROVED** [review]

**Comment by iblis17** [comment]
FreeBSD CI builds failed after this PR merged https://julia.iblis.cnmc.tw/#/builders/1/builds/25

**Comment by rfourquet** [comment]
Ouch, it seems the `len` field of `String` disappeared in the last month... maybe 2b47865a7 ? it seems `sizeof` is the alternative, will make a patch.

File: base/gmp.jl
Comment (iblis17) [inline base/gmp.jl]: The stacktrace point out this line

Comment (yuyichao) [inline base/gmp.jl]: Good that the len field is removed....

**Comment by tkelman** [comment]
Before merging, should really check that the CI result isn't so old as to be obsolete when a PR hasn't had any recent activity.

**Comment by StefanKarpinski** [comment]
Yeah, sorry about that.

**Comment by timholy** [comment]
Unless we want to re-run every PR every 24 hrs, this stuff is just going to happen sometimes.

Best lesson is to merge good work quickly. We need an army of @tkelman(s) with their insightful eyes on everything all the time :smile:.

**Comment by tkelman** [comment]
We can re-run the ones that are ready to merge before doing so, if it's been more than a few days since CI last ran.

**Comment by KristofferC** [comment]
This either has to be automated a la Rust or it will keep happening.

**Comment by tkelman** [comment]
homu does not support appveyor https://github.com/barosl/homu/issues/87

for now please put a bit of thought into how old the CI results are before you merge something.

**Pull request merged**
